File: 프로그래머스/0/120894. 영어가 싫어요/영어가 싫어요.cs
using System;
using System.Collections.Generic;

public class Solution
{
    public long solution(string numbers)
    {
        string[] zero_to_nine = new string[] 
        {"zero","one","two","three","four","five","six","seven","eight","nine"};
        
        string[] int_zero_to_nine = new string[] 
        {"0","1","2","3","4","5","6","7","8","9"};
        
        for(int i=0; i<zero_to_nine.Length; i++){
            if(numbers.Contains(zero_to_nine[i]))
            numbers = numbers.Replace(zero_to_nine[i],int_zero_to_nine[i]);   
        }
        long answer = long.Parse(numbers);
        return answer;
    }
}

File: 프로그래머스/0/120894. 영어가 싫어요/영어가 싫어요.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

long long solution(string numbers) 
{
    vector<string> numberStr = {"zero","one","two","three","four","five","six","seven","eight","nine"};
    vector<string> number = {"0","1","2","3","4","5","6","7","8","9"};
    long long answer = 0;
    
     for (int i = 0; i < numberStr.size(); i++) 
     {
        // 모든 위치 찾기
        size_t pos = numbers.find(numberStr[i]);
        while (pos != string::npos) 
        {
            numbers.replace(pos, numberStr[i].length(), number[i]);
            pos = numbers.find(numberStr[i], pos + 1); // 다음 위치를 찾기
        }
     }
    return stoll(numbers);
}